### PR TITLE
[ARO-7788] Don't overwrite default installer version in local dev

### PR DESCRIPTION
**Pull request by mrWinston**
### Which issue this PR addresses:
Fixes https://issues.redhat.com/browse/ARO-7788

### What this PR does / why we need it:

Currently, the dev cluster creation tooling always makes sure that the default openshift version together with the default installer image is written to the cosmos db before creating the cluster. This however makes it impossible to use a custom installer image with the default openshift version, as any modifications to the installer image are overwritten.

This PR changes this behaviour so that the default version and installer image is only written when it is not already present.

### Test plan for issue:

- testing was performed according to instructions in jira ticket.

### Is there any documentation that needs to be updated for this PR?

- No
- 
### How do you know this will function as expected in production? 

- this doesn't affect production.

### CODE REVIEW

File: pkg/util/cluster/cluster.go [hunk 1]
```diff
@@ -484,6 +484,16 @@ func (c *Cluster) createCluster(ctx context.Context, vnetResourceGroup, clusterN
...
 
+var insecureLocalClient *http.Client = &http.Client{
+	Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	},
+}
+
+var localDefaultURL string = "https://localhost:8443"
```
Comment (mociarain) [inline pkg/util/cluster/cluster.go:495]: What `var` and not `const`?

File: pkg/util/cluster/cluster.go [hunk 2]
```diff
@@ -501,30 +511,68 @@ func (c *Cluster) registerSubscription(ctx context.Context) error {
...
 	if err != nil {
 		return err
 	}
 
 	return resp.Body.Close()
 }
 
-func (c *Cluster) insertDefaultVersionIntoCosmosdb(ctx context.Context) error {
+// getVersionsInCosmosDB connects to the local RP endpoint and queries the
+// available OpenShiftVersions
```
Comment (mociarain) [inline pkg/util/cluster/cluster.go:530]: Pure curiosity:

I'm still new to go and one of the things I find very odd is the lack of recommended line lengths. Here you seem to have tightened your comments to ~80 but longer in the code. Is there a reason for this/some convention I'm unaware of?

File: pkg/util/cluster/cluster.go [hunk 3]
```diff
@@ -501,30 +511,68 @@ func (c *Cluster) registerSubscription(ctx context.Context) error {
...
+}
+
+// ensureDefaultVersionInCosmosdb puts a default openshiftversion into the
+// cosmos DB IF it doesn't already contain an entry for the default version. It
+// is hardcoded to use the local-RP endpoint `https://localhost:8443`
+//
+// It returns without an error when a default version is already present or a
+// default version was successfully put into the db.
+func (c *Cluster) ensureDefaultVersionInCosmosdb(ctx context.Context) error {
+	// first, we make sure that we don't overwrite any existing entry for the default version
```
Comment (mociarain) [inline pkg/util/cluster/cluster.go:562]: Do we need this comment?

File: pkg/util/cluster/cluster.go [hunk 4]
```diff
@@ -501,30 +511,68 @@ func (c *Cluster) registerSubscription(ctx context.Context) error {
...
+	versionsInDB, err := getVersionsInCosmosDB(ctx)
+	if err != nil {
+		return fmt.Errorf("couldn't query versions in cosmosdb: %w", err)
+	}
+
+	for _, versionFromDB := range versionsInDB {
+		// we already have an entry for the default version
```
Comment (mociarain) [inline pkg/util/cluster/cluster.go:569]: Do we need this comment?

**Review by mociarain — COMMENTED** [review]
Nice change. Just a few questions

File: pkg/util/cluster/cluster.go (hunk 1 above)
Comment (mrWinston) [reply pkg/util/cluster/cluster.go:495]: Good point, `const` would indeed be the better choice here.

File: pkg/util/cluster/cluster.go (hunk 2 above)
Comment (mrWinston) [reply pkg/util/cluster/cluster.go:530]: No, i don't think there's a convention regarding that. I wrap a line when it feels like it's getting too long, and for godoc comments i just like the look of them them being wrapped :)

File: pkg/util/cluster/cluster.go (hunk 3 above)
Comment (mociarain) [reply pkg/util/cluster/cluster.go]: No worries. I'm borderline anti-comment so I challenge them often. Been bitten by stale comments too many times in the past :D

**Review by mociarain — APPROVED** [review]

**Review by tiguelu — APPROVED** [review]

**Review by tiguelu — CHANGES_REQUESTED** [review]
@mrWinston Sorry, E2E cannot run in the PR because the source branch is in a fork of the ARO-RP. Unfortunately, we have this restriction for security reasons, so you will need to create a new PR using a branch created directly in the ARO-RP repo.

I will close this PR.

Sorry for that, and thank you for fixing this bug!

**Pull request closed without merging**